src/com/intellij/idea/plugin/hybris/system/java/codeInsight/hints/LoggerInlayHintsProvider.kt

```diff
@@ -0,0 +1,178 @@
+    override val defaultAnchor: CodeVisionAnchorKind
+        get() = CodeVisionAnchorKind.Default
+    override val id: String
+        get() = "LoggerInlayHintsProvider"
+    override val name: String
+        get() = "SAP Commerce Cloud Logger"
+
+    override val relativeOrderings: List<CodeVisionRelativeOrdering>
+        get() = emptyList()
+
+    override fun computeLenses(editor: Editor, psiFile: PsiFile): List<Pair<TextRange, CodeVisionEntry>> {
+        //todo check it is hybris
+
+        val entries = mutableListOf<Pair<TextRange, CodeVisionEntry>>()
+
+        psiFile.accept(object : PsiRecursiveElementVisitor() {
+            override fun visitElement(element: PsiElement) {
+                super.visitElement(element)
+                val targetElement = when (element) {
+                    is PsiClass -> {
+                        val psiKeyword = PsiTreeUtil.getChildrenOfType(element, PsiKeyword::class.java)?.first()?.text
+                        if (psiKeyword == "class")
+                             element.nameIdentifier
+                        else null
+                    }
+                    is PsiPackageStatement -> {
+                        val psiKeyword = PsiTreeUtil.getChildrenOfType(element, PsiKeyword::class.java)?.first()?.text
+                        if (psiKeyword == "package")
+                            element.packageReference
+                        else null
+                    }
+                    else -> null
+                }
+                if (targetElement == null) return
+
+                val loggerIdentifier = extractIdentifierForLogger(element, psiFile) ?: return
+
+                val handler = ClickHandler(targetElement, loggerIdentifier)
+                val range = InlayHintsUtils.getTextRangeWithoutLeadingCommentsAndWhitespaces(targetElement)
+                entries.add(range to ClickableTextCodeVisionEntry("log level", id, handler, HybrisIcons.Log.TOGGLE, "", "Setup the logger for SAP Commerce Cloud"))
+            }
+        })
+
+        return entries
+    }
+
+    fun extractIdentifierForLogger(element: PsiElement, file: PsiFile): String? = when (element) {
+        is PsiClass -> file.packageName()?.let { "$it.${element.name}" }
+        is PsiPackageStatement -> element.packageName
+        else -> null
+    }
+
+    private inner class ClickHandler(
+        element: PsiElement,
+        private val loggerIdentifier: String,
+    ) : (MouseEvent?, Editor) -> Unit {
+        private val elementPointer = SmartPointerManager.createPointer(element)
+
+        override fun invoke(event: MouseEvent?, editor: Editor) {
+            if (isInlaySettingsEditor(editor)) return
+            val element = elementPointer.element ?: return
+            handleClick(editor, element, loggerIdentifier)
+        }
+    }
+
+    fun handleClick(editor: Editor, element: PsiElement, loggerIdentifier: String) {
+        val actionGroup = DefaultActionGroup().apply {
```

Comment:
Define new action group and action via `xml`, later in the code get that action group via `ActionManager.getInstance().getAction(..)`. See example - `ccv2.toolbar.actions`.

Move logic responsible for availability of the action to the action itself, its presentation configuration set via `update` method. Rely on `dataContext` to pass required data to be used in the action itself.

p.s. it can stay as a first start, but it will produce too many unwanted instances of the Actions.